jax/interpreters/xla.py

```diff
@@ -463,22 +464,88 @@ def jaxpr_collectives(jaxpr):
 
 ### xla_call underlying jit
 
-def _xla_call_impl(fun: lu.WrappedFun, *args, device, backend, name):
-  compiled_fun = _xla_callable(fun, device, backend, name, *map(arg_spec, args))
+def _xla_call_impl(
+    fun: lu.WrappedFun,
+    *args,
+    device: Optional[xe.Device],
+    backend,
+    name: str,
+    donate: Sequence[bool],
+):
+  if len(args) != len(donate):
```

Comment:
Perhaps @mattjj has input here? This feels a bit strange to me.

Reply:
Are you worried about the grad-of-jit case? I think it should error before it hits this. Maybe add a test case to make sure it errors nicely?

Reply:
With @mattjj's help this is now resolved. properly 😄